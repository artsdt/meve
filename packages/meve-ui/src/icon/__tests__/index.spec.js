import Icon from '..'
import { mount } from '@vue/test-utils'

test('test icon size', () => {
  const wrapper = mount(Icon, {
    propsData: {
      name: 'checkbox-marked-circle',
      size: 30,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test icon color', () => {
  const wrapper = mount(Icon, {
    propsData: {
      name: 'checkbox-marked-circle',
      color: 'red',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test icon on click', async () => {
  const handleClick = jest.fn()

  const wrapper = mount(Icon, {
    propsData: {
      name: 'checkbox-marked-circle',
    },
    context: {
      on: {
        click: handleClick,
      },
    },
  })

  await wrapper.trigger('click')
  expect(handleClick).toHaveBeenCalledTimes(1)

  wrapper.destroy()
})
