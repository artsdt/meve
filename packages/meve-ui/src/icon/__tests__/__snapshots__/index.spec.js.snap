// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test icon color 1`] = `"<i class=\\"m-icon m--box meve-icon--set meve-icon-checkbox-marked-circle\\" style=\\"color: red;\\"></i>"`;

exports[`test icon size 1`] = `"<i class=\\"m-icon m--box meve-icon--set meve-icon-checkbox-marked-circle\\" style=\\"font-size: 30px;\\"></i>"`;
